cart/infrastructure/placeorder/logger.go

```diff
@@ -0,0 +1,155 @@
+package email
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	cartDomain "flamingo.me/flamingo-commerce/v3/cart/domain/cart"
+	"flamingo.me/flamingo-commerce/v3/cart/domain/placeorder"
+	authDomain "flamingo.me/flamingo/v3/core/oauth/domain"
+	"flamingo.me/flamingo/v3/framework/flamingo"
+	"fmt"
+	"golang.org/x/mod/modfile"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+)
+
+type (
+	// PlaceOrderLoggerAdapter provides an implementation of the Service as email adapter
+	PlaceOrderLoggerAdapter struct {
+		useFlamingoLog bool
+		logAsFile      bool
+		logDirectory   string
+		logger         flamingo.Logger
+	}
+)
+
+var (
+	_ placeorder.Service = new(PlaceOrderLoggerAdapter)
+)
+
+// Inject dependencies
+func (e *PlaceOrderLoggerAdapter) Inject(logger flamingo.Logger,
+	config *struct {
+		UseFlamingoLog bool   `inject:"config:commerce.cart.placeOrderLogger.useFlamingoLog,optional"`
+		LogAsFile      bool   `inject:"config:commerce.cart.placeOrderLogger.logAsFile,optional"`
+		LogDirectory   string `inject:"config:commerce.cart.placeOrderLogger.logDirectory,optional"`
+	}) {
+	e.logger = logger.WithField("module", "cart").WithField("category", "emailAdapter")
+	if config != nil {
+		e.useFlamingoLog = config.UseFlamingoLog
+		e.logAsFile = config.LogAsFile
+		e.logDirectory = config.LogDirectory
+	}
+}
+
+// PlaceGuestCart places a guest cart as order email
+func (e *PlaceOrderLoggerAdapter) PlaceGuestCart(ctx context.Context, cart *cartDomain.Cart, payment *placeorder.Payment) (placeorder.PlacedOrderInfos, error) {
+	err := e.checkPayment(cart, payment)
+	if err != nil {
+		return nil, err
+	}
+	err = e.logOrder(cart, payment)
+	if err != nil {
+		return nil, err
+	}
+	var placedOrders placeorder.PlacedOrderInfos
+	placedOrders = append(placedOrders, placeorder.PlacedOrderInfo{
+		OrderNumber: cart.ID,
+	})
+
+	return placedOrders, nil
+}
+
+// PlaceCustomerCart places a customer cart as order email
+func (e *PlaceOrderLoggerAdapter) PlaceCustomerCart(ctx context.Context, auth authDomain.Auth, cart *cartDomain.Cart, payment *placeorder.Payment) (placeorder.PlacedOrderInfos, error) {
```

Comment:
we could try to reduce this code duplication

Reply:
good point